### PR TITLE
[ornaments] Honor updates to the ornament options

### PR DESCRIPTION
**Pull request by nishant-karajgikar**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes https://github.com/mapbox/mapbox-maps-ios/issues/31

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog>Fix issue where updates to ornament options were not honored</changelog>`.

### Summary of changes

This PR ensures that updates to ornament options are honored. The bug was in the `updateOrnaments` method where the MapView was incorrectly re-using the old ornament options instead of the new one.

### CODE REVIEW

**Review by macdrevx — COMMENTED** [review]
Can we add a unit test?

**Review by neelmistry94 — APPROVED** [review]

**Comment by nishant-karajgikar** [comment]
Yep added an integration tests that checks for updates across all the managers.

File: Mapbox/MapboxMapsTests/OptionsIntegrationTests.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import XCTest
+@testable import MapboxMaps
+import CoreLocation
+
+internal class OptionsIntegrationTest: MapViewIntegrationTestCase {
+
+    internal func testBaseClass() throws {
+        // Do nothing
+    }
+
```
Comment (macdrevx) [inline Mapbox/MapboxMapsTests/OptionsIntegrationTests.swift:10]: What is the purpose of this?

File: Mapbox/MapboxMapsTests/OptionsIntegrationTests.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        newOptions.ornaments.showsScale = false
+        newOptions.ornaments.showsCompass = false
+
+        mapView.update { (options) in
+            options = newOptions
+        }
+
+        XCTAssert(mapView.gestureManager.gestureOptions == newOptions.gestures)
+        XCTAssert(mapView.cameraManager.mapCameraOptions == newOptions.camera)
+        XCTAssert(mapView.locationManager.locationOptions == newOptions.location)
```
Comment (macdrevx) [inline Mapbox/MapboxMapsTests/OptionsIntegrationTests.swift:30]: Could these use `XCTAssertEqual`?

File: Mapbox/MapboxMapsTests/OptionsIntegrationTests.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        XCTAssert(!mapView.ornamentsManager.ornamentConfig.ornaments.contains(where: {$0.type == .compass
+                                                                                || $0.type == .mapboxScaleBar}))
```
Comment (macdrevx) [inline Mapbox/MapboxMapsTests/OptionsIntegrationTests.swift:32]: Could this use trailing closure syntax?

```suggestion
        XCTAssert(!mapView.ornamentsManager.ornamentConfig.ornaments.contains {
            $0.type == .compass || $0.type == .mapboxScaleBar
        })
```

File: Mapbox/MapboxMapsTests/OptionsIntegrationTests.swift (hunk 1 above)
Comment (nishant-karajgikar) [reply Mapbox/MapboxMapsTests/OptionsIntegrationTests.swift:10]: ah left this in by mistake!

**Review by macdrevx — APPROVED** [review]

**Pull request merged**